lightning/src/ln/msgs.rs

```diff
@@ -1515,6 +1599,186 @@ impl_writeable_len_match!(NodeAnnouncement, {
 	contents
 });
 
+impl Readable for QueryShortChannelIds {
+	fn read<R: Read>(r: &mut R) -> Result<Self, DecodeError> {
+		let chain_hash: BlockHash = Readable::read(r)?;
+
+		// We expect the encoding_len to always includes the 1-byte
+		// encoding_type and that short_channel_ids are 8-bytes each
+		let encoding_len: u16 = Readable::read(r)?;
+		if encoding_len == 0 || (encoding_len - 1) % 8 != 0 {
+			return Err(DecodeError::InvalidValue);
+		}
+
+		// Must be encoding_type=0 uncompressed serialization. We do not
+		// support encoding_type=1 zlib serialization.
+		let encoding_type: u8 = Readable::read(r)?;
+		if encoding_type != EncodingType::Uncompressed as u8 {
+			return Err(DecodeError::InvalidValue);
+		}
+
+		// Read short_channel_ids (8-bytes each), for the encoding_len
+		// less the 1-byte encoding_type
+		let mut short_channel_ids = vec![];
+		let mut read_remaining = encoding_len - 1;
+		while read_remaining > 0 {
```

Comment:
Why not `for _ in 0..(encoding_len - 1)/8`?

Reply:
Yes! I'm still getting used to replacing c-style `for` loops with `for in range` haha